src/galaxy/CustomSystem.cpp

```diff
@@ -660,14 +752,189 @@ void CustomSystemsDatabase::Load()
 
 	RegisterCustomSystemsAPI(L);
 
-	LUA_DEBUG_CHECK(L, 0);
+	LUA_DEBUG_END(L, 0);
+	return L;
+}
+
+void CustomSystemsDatabase::Load()
+{
+	PROFILE_SCOPED()
+
+	LoadAllLuaSystems();
+
+	// Load Json array files containing random-fill system definitions
+	std::string partialPath = FileSystem::JoinPathBelow(m_customSysDirectory, "partial");
+	for (auto &file : FileSystem::gameDataFiles.Recurse(partialPath)) {
+		if (!ends_with_ci(file.GetPath(), ".json"))
+			continue;
+
+		PROFILE_SCOPED_DESC("Load Partial System List")
+		const Json fileData = JsonUtils::LoadJsonDataFile(file.GetPath());
+		for (const Json &sysdef : fileData) {
+			if (!sysdef.is_object())
+				continue;
+
+			LoadSystemFromJSON(file.GetPath(), sysdef);
+		}
+	}
+
+	// Load top-level custom system defines
+	for (auto &file : FileSystem::gameDataFiles.Enumerate(m_customSysDirectory, 0)) {
+		if (!ends_with_ci(file.GetPath(), ".json"))
+			continue;
+
+		LoadSystemFromJSON(file.GetPath(), JsonUtils::LoadJsonDataFile(file.GetPath()));
+	}
+
+	// Load all complete custom system definitions
+	std::string customPath = FileSystem::JoinPathBelow(m_customSysDirectory, "custom");
+	for (auto &file : FileSystem::gameDataFiles.Recurse(customPath)) {
+		if (!ends_with_ci(file.GetPath(), ".json"))
+			continue;
+
+		LoadSystemFromJSON(file.GetPath(), JsonUtils::LoadJsonDataFile(file.GetPath()));
+	}
+}
+
+void CustomSystemsDatabase::LoadAllLuaSystems()
+{
+	PROFILE_SCOPED()
+
+	assert(!s_activeCustomSystemsDatabase);
+	s_activeCustomSystemsDatabase = this;
```

Comment:
Maybe we could explicitly pass this object to Lua? Otherwise it looks a little suspicious.

Reply:
Were we intending to continue supporting lua system definitions long-term I'd consider it; however I don't intend to sink any additional time into supporting lua custom system definitions and consider all lua-related code deprecated.